code/game/objects/effects/effect_system/effects_foam.dm

```diff
@@ -23,6 +23,8 @@
 	/turf/open/chasm,
 	/turf/open/lava))
 	var/slippery_foam = TRUE
+	var/one_apply_per_object = FALSE
+	var/list/applied_atoms
```

Comment:
clear your list on destroy please, don't drag around a list of references to this many objects